rtloader/test/aggregator/aggregator.go

```diff
@@ -238,6 +238,6 @@ func submitHistogramBucket(id *C.char, cMetricName *C.char, cVal C.longlong, cLo
 //export submitEventPlatformEvent
 func submitEventPlatformEvent(id *C.char, _rawEvent *C.char, _eventType *C.char) {
 	checkID = C.GoString(id)
-	rawEvent = C.GoString(_rawEvent)
+	rawEvent = []byte(C.GoString(_rawEvent))
```

Comment:
Not sure how hard it would be, but would be nice to use `GoBytes` instead to avoid double allocation here (there was another place with this pattern).

Reply:
Indeed, this pattern is also present in `pkg/collector/python/aggregator.go`.

I also initially wanted to switch to `C.GoBytes`, but its different signature requires some more work:
```go
// C data with explicit length to Go []byte
func C.GoBytes(unsafe.Pointer, C.int) []byte
```

The size can be obtained with `strlen`, but it returns a `size_t` whereas `GoBytes` expects an `int`.

So, in the end, here is what the use of `GoBytes` would look like:

```diff
diff --git a/pkg/collector/python/aggregator.go b/pkg/collector/python/aggregator.go
index 8f869b9949..15bfe50731 100644
--- a/pkg/collector/python/aggregator.go
+++ b/pkg/collector/python/aggregator.go
@@ -9,6 +9,8 @@
 package python
 
 import (
+       "unsafe"
+
        "github.com/DataDog/datadog-agent/pkg/aggregator"
        chk "github.com/DataDog/datadog-agent/pkg/collector/check"
        "github.com/DataDog/datadog-agent/pkg/metrics"
@@ -16,6 +18,7 @@ import (
 )
 
 /*
+#include <string.h>
 #include <datadog_agent_rtloader.h>
 #cgo !windows LDFLAGS: -L${SRCDIR}/../../../rtloader/build/rtloader -ldatadog-agent-rtloader -ldl
 #cgo windows LDFLAGS: -L${SRCDIR}/../../../rtloader/build/rtloader -ldatadog-agent-rtloader -lstdc++ -static
@@ -149,5 +152,5 @@ func SubmitEventPlatformEvent(checkID *C.char, rawEvent *C.char, eventType *C.ch
                log.Errorf("Error submitting event platform event to the Sender: %v", err)
                return
        }
-       sender.EventPlatformEvent([]byte(C.GoString(rawEvent)), C.GoString(eventType))
+       sender.EventPlatformEvent(C.GoBytes(unsafe.Pointer(rawEvent), C.int(C.strlen(rawEvent))), C.GoString(eventType))
 }
```

I eventually thought that the code with `C.GoBytes(…)` was less readable than the one with the `[]byte` cast.

But I can switch again if you still think it’s better.
WDYT ?

Reply:
AIUI ep events may be especially large for some checks, so maybe let's go with the less readable version to avoid large copies.

Better still would be to get the length directly from Python by using `s#` [here](https://github.com/datadog/datadog-agent/blob/15b0ad8d282857659fdfa83b578d220ff321bcd8/rtloader/common/builtins/aggregator.c#L431) and get rid of `strlen` too.

Reply:
Thanks @vickenty for the tip about `s#`: it makes things much smarter!
Here are the changes I did to use `GoBytes(…)`: https://github.com/DataDog/datadog-agent/pull/16084/commits/17b0450a72985c1e6973c496b877744c55b07e18.
It addresses the two locations with the same pattern.